### ISSUE TITLE
`too-few-function-args` overlaps with `no-value-for-parameter`

### ISSUE THREAD

**Issue by rogersheu** [pylint-dev/pylint#9881]
### Bug description

I realized that the addition of a new message, `too-few-function-args`, in the recent PR #9847, overlaps with an existing message `no-value-for-parameter`.

```python
    "E1120": (
        "No value for argument %s in %s call",
        "no-value-for-parameter",
        "Used when a function call passes too few arguments.",
    ),
```

```python
    "E1145": (
        "Too few positional arguments for %s call",
        "too-few-function-args",
        "Used when a function or method call has fewer arguments than expected.",
    ),
```

Since `no-value-for-parameter` was the incumbent, it makes sense to keep it, unless we collectively decide otherwise. It may also be worth revisiting the contents/format of the message output by this check.

### Command used

```shell
pylint "**/*.py"
```

### Pylint output

```python
************* Module isinstances
isinstances.py:4:0: E1145: Too few positional arguments for function call (too-few-function-args)
isinstances.py:5:0: E1145: Too few positional arguments for function call (too-few-function-args)
```

### Expected behavior

```python
************* Module isinstances
isinstances.py:4:0: E1120: No value for argument '__class_or_tuple' in function call (no-value-for-parameter)
isinstances.py:5:0: E1120: No value for argument '_obj' in function call (no-value-for-parameter)
isinstances.py:5:0: E1120: No value for argument '__class_or_tuple' in function call (no-value-for-parameter)
```

### Pylint version

```shell
pylint 3.3.0.dev0
```

```[tasklist]
### Tasks
```
